studio/pages/project/[ref]/functions/[functionSlug]/index.tsx

```diff
@@ -26,32 +29,95 @@ const CHART_INTERVALS: ChartIntervals[] = [
+  {
+    key: '1hr',
+    label: '1 hour',
+    startValue: 1,
+    startUnit: 'hour',
+    format: 'MMM D, h:mma',
+  },
+  {
+    key: '1day',
+    label: '1 day',
+    startValue: 1,
+    startUnit: 'hour',
+    format: 'MMM D, h:mma',
+  },
+  {
+    key: '7day',
+    label: '7 days',
+    startValue: 7,
+    startUnit: 'day',
+    format: 'MMM D',
+  },
 ]
 
 const PageLayout: NextPageWithLayout = () => {
   const router = useRouter()
   const { ref: projectRef, functionSlug } = useParams()
   const [interval, setInterval] = useState<string>('15min')
   const selectedInterval = CHART_INTERVALS.find((i) => i.key === interval) || CHART_INTERVALS[1]
-  const { data: selectedFunction } = useEdgeFunctionQuery({ projectRef, slug: functionSlug })
-  const id = selectedFunction?.id
-
-  const { data, error } = useFunctionsInvStatsQuery({
+  const { data: selectedFunction } = useEdgeFunctionQuery({
     projectRef,
-    functionId: id,
-    interval: selectedInterval.key,
+    slug: functionSlug,
   })
-  const isChartLoading = !data?.result && !error ? true : false
-  const normalizedData = useMemo(() => {
-    return (data?.result || []).map((d: any) => ({
+  const id = selectedFunction?.id
+  const resourceUsageMetricsEnabled = useFlag('enableResourceUsageMetricsForEdgeFunctions')
+
+  const { data: invStatsData, error: invStatsError } = useFunctionsInvStatsQuery(
+    {
+      projectRef,
+      functionId: id,
+      interval: selectedInterval.key,
+    },
+    { enabled: !resourceUsageMetricsEnabled }
+  )
+
+  const { data: reqStatsData, error: reqStatsError } = useFunctionsReqStatsQuery(
+    {
+      projectRef,
+      functionId: id,
+      interval: selectedInterval.key,
+    },
+    { enabled: resourceUsageMetricsEnabled }
+  )
+
+  const { data: resourceUsageData, error: resourceUsageError } = useFunctionsResourceUsageQuery(
+    {
+      projectRef,
+      functionId: id,
+      interval: selectedInterval.key,
+    },
+    { enabled: resourceUsageMetricsEnabled }
+  )
+
+  const isInvStatsChartLoading = !invStatsData || (!invStatsData.result && !invStatsError)
+  const isReqStatsChartLoading = !reqStatsData || (!reqStatsData?.result && !reqStatsError)
+  const isResourceUsageChartLoading =
+    !resourceUsageData || (!resourceUsageData?.result && !resourceUsageError)
+
+  const normalizedReqStatsData = useMemo(() => {
+    const result = resourceUsageMetricsEnabled ? reqStatsData?.result : invStatsData?.result
+    return (result || []).map((d: any) => ({
       ...d,
       timestamp: isUnixMicro(d.timestamp) ? unixMicroToIsoTimestamp(d.timestamp) : d.timestamp,
     }))
-  }, [data?.result])
+  }, [invStatsData?.result, reqStatsData?.result, resourceUsageMetricsEnabled])
+
+  const normalizedResourceUsageData = useMemo(() => {
+    return (resourceUsageData?.result || []).map((d: any) => ({
+      avg_cpu_time_used: d.avg_cpu_time_used,
+      avg_memory_used: (d.avg_heap_memory_used + d.avg_external_memory_used) / (1024 * 1024),
```

Comment:
this logic can't be pushed to the sql query?

Reply:
moved to the sql query